### ISSUE TITLE
Implementation of class weights with tf.nn.softmax_cross_entropy_with_logits_v2

### ISSUE THREAD

**Issue by bipinkc19** [tensorflow/tensorflow#27741]
**System information**
- TensorFlow version (you are using): 1.12.0
- Are you willing to contribute it (Yes/No): Yes, if my skills are enough.

__Requested Feature__
Currently while implementing weighted loss there is tf.nn.weighted_cross_entropy_with_logits which uses sigmoid so it is restricted to binary classification. For multiclass classification using softmax categorical cross entropy, there isn't any feature.

Also in many forums, while implementing the class weights people often suggest something like this:
```python
weighted_logits = tf.multiply(weights, logits)
loss = tf.nn.softmax_cross_entropy_with_logits_v2
```
Is this the correct method? Because we are messing around with the output of the model, while we should have been messing around with the calculated [batch_size, num_class] shape loss tensor.

I think this is the correct way:
```python
        y_hat_softmax = tf.nn.softmax(logits)
        y_cross = y * tf.log(y_hat_softmax)
        weighted_cross = tf.multiply(class_weight, y_cross)
        weighted_cross_reduced = - tf.reduce_sum(weighted_cross, 1)
        loss = tf.reduce_mean(weighted_cross_reduced)
```
Here we are changing the y_cross which should be actually changed, which then changes the gradient flow. But changing logits (by multiplying with weight) will make the model understand that the correct output is different thus biasing result.

eg. If a model has three classes and the output logit is [1, 4, 2] then it is making the correct decision if the actual class is the second one. But multiplying with weight [1, 1, 4] will make the result [1, 4, 8] thus saying that the third logit should have been high to the loss function, thus completely changing the supervision in supervised learning. But actually, we should have been messing around with the loss by scaling the loss according to the weights. 

**Will this change the current api? How?**
Don't know.

**Who will benefit with this feature?**
Whoever is implementing this incorrectly will benefit from this, democratizing Machine Learning.

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose). Thanks!

**Comment by rmothukuru** [comment]
@bipinkc19,
Sorry for the delayed response. Can you please let us know if [weighted_cross_entropy_with_logits](https://www.tensorflow.org/api_docs/python/tf/nn/weighted_cross_entropy_with_logits) is the functionality that you are looking for? Thanks!
